ports/javascript/Makefile

```diff
@@ -16,7 +16,10 @@ INC += -I$(BUILD)
 CPP = clang -E
 
 ifdef EMSCRIPTEN
-    CPP += -isystem $(EMSCRIPTEN)/system/include/libc -cxx-isystem $(EMSCRIPTEN)/system/include/libcxx
+	CPP += -D__EMSCRIPTEN__
+	CPP += --sysroot $(EMSCRIPTEN)/system
+	CPP += -isystem $(EMSCRIPTEN)/system/include/libc
+	CPP += -cxx-isystem $(EMSCRIPTEN)/system/include/libcxx
```

Comment:
@dpgeorge @pmp-p

I've finally discovered that there is a slightly better way of handling these :

```
CPP += $(addprefix -isystem, $(shell $(CC) -E -x c++ /dev/null -v 2>&1 |sed -e '/^\#include <...>/,/^End of search/{ //!b };d'))
```

By parsing the verbose output of `emcc` it's possible to retrieve the include paths.

Reply:
Very nice ! i like that it's a very complete output,

@embeddedt  one thing though : it seems SDL include folder looks like a ```-s USE_SDL=1``` (1.3.0) instead of a 2

also maybe add LC_ALL=C just in case ```End of search``` gets a localized version ?

Reply:
To fix that the `JSFLAGS` line should be moved above `CC = emcc -g4` and the new line is this:

```
$(addprefix -isystem, $(shell env LC_ALL=C $(CC) -E -x c++ /dev/null -v 2>&1 |sed -e '/^\#include <...>/,/^End of search/{ //!b };d'))
```

Reply:
Confirmed that the hack is not needed.

Reply:
it works very well when adding specific -s USE_flags that are normally in CFLAGS but should not be passed to clang -E. 

 i see a reason for user mod to have separate CFLAGS and CPPFLAGS
or some more magic in the makefile to remove the  -s * pattern from the user mod flags ?

Reply:
What is the problem? It is compiling fine without any errors about unknown options.

Reply:
@embeddedt you're right, i missed that user flags include will get translated to -isystem ones too

i'll try to enhance that PR for user mod system !

Reply:
Ideally we also want to pass proper predefined macros to `clang` as well; I'm working on another one of these parsing hacks that should do that. That way we can avoid this ugliness (which will fall apart on a different architecture):

https://github.com/littlevgl/lv_micropython/blob/1381be71fb2f11c6e0841b2beebf1f3aa84bcba2/ports/javascript/Makefile#L26

Reply:
Alright, I can now pass macros to `clang`; see my latest comments on the `Makefile` (near `CPP = clang -E`).